### PR TITLE
snapshot-timestamp for asraa

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Documentation

**TO VERIFY:**
```
REPO=$(pwd)/ceremony/2022-07-12  ./scripts/verify.sh 321
```

You should see
`timestamp.json` and `snapshot.json` validly signed under staged metadata, with versions 38.

You can also manually verify the hashes.
for.e.g

```
$ sha256sum $(pwd)/ceremony/2022-07-12/staged/targets.json
730ce11e76f9c91f3814556f97f84566d95f6908b246a0013d7d66db52167117  /home/asraa/git/root-signing/ceremony/2022-07-12/staged/targets.json
$  cat $(pwd)/ceremony/2022-07-12/staged/snapshot.json | jq -r '.signed.meta."targets.json"'
{
  "hashes": {
    "sha256": "730ce11e76f9c91f3814556f97f84566d95f6908b246a0013d7d66db52167117",
    "sha512": "7c6d8beacd11aa9011a700a0509fead75fcf54d836a414f5633a2c29ee3c10415797a38aa85396953a3c6b206a052225852ce2cc85ae8b110d849855d32d0f59"
  },
  "length": 5984,
  "version": 4
}


```

### CODE REVIEW

**Comment by asraa** [comment]
cc @elfotografo007 would love review!

File: ceremony/2022-07-12/staged/snapshot.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+				"length": 783,
+				"version": 1
+			},
+			"root.json": {
+				"hashes": {
+					"sha256": "f5ad897c9414cca99629f400ac3585e41bd8ebb44c5af07fb08dd636a9eced9c",
+					"sha512": "7445ddfdd338ef786c324fc3d68f75be28cb95b7fb581d2a383e3e5dde18aa17029a5636ec0a22e9631931bbcb34057788311718ea41e21e7cdd3c0de13ede42"
+				},
+				"length": 5297,
+				"version": 2
```
Comment (asraa) [inline ceremony/2022-07-12/staged/snapshot.json:34]: note the bug here: https://github.com/sigstore/root-signing/pull/320#discussion_r920174081

it won't affect clients. tuf clients do not need to check for `root.json` in the snapshot anymore. https://github.com/theupdateframework/go-tuf/issues/203

**Comment by kommendorkapten** [comment]
Looks good:

```
kommendorkapten@m1m14:~/git/root-signing % GITHUB_USER=kommendorkapten REPO=$(pwd)/ceremony/2022-07-12 ./scripts/verify.sh 321
+ '[' -z kommendorkapten ']'
+ '[' -z /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12 ']'
+ git checkout main
Switched to branch 'main'
+ git status
On branch main
nothing to commit, working tree clean
+ git remote -v
origin	git@github.com:kommendorkapten/root-signing.git (fetch)
origin	git@github.com:kommendorkapten/root-signing.git (push)
upstream	git@github.com:sigstore/root-signing.git (fetch)
upstream	git@github.com:sigstore/root-signing.git (push)
+ git remote rm upstream
+ git remote add upstream git@github.com:sigstore/root-signing.git
+ git remote rm origin
+ git remote add origin git@github.com:kommendorkapten/root-signing.git
+ git remote -v
origin	git@github.com:kommendorkapten/root-signing.git (fetch)
origin	git@github.com:kommendorkapten/root-signing.git (push)
upstream	git@github.com:sigstore/root-signing.git (fetch)
upstream	git@github.com:sigstore/root-signing.git (push)
+ go build -o verify ./cmd/verify
+ '[' -f piv-attestation-ca.pem ']'
+ [[ ! -z 321 ]]
+ echo 'Pull Request: 321'
Pull Request: 321
+ git branch -D VERIFY
Deleted branch VERIFY (was e9f113f).
+ git fetch upstream pull/321/head:VERIFY
From github.com:sigstore/root-signing
 * [new ref]         refs/pull/321/head -> VERIFY
+ git checkout VERIFY
Switched to branch 'VERIFY'
+ '[' -d /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12/keys ']'
+ ./verify keys --root piv-attestation-ca.pem --key-directory /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12/keys

Outputting key verification and OpenSSL commands...

VERIFIED KEY WITH SERIAL NUMBER 13078778
	TUF key id: 2f64fb5eac0cf94dd39bb45308b98920055e9a0d8e012a7220787834c60aef97

VERIFIED KEY WITH SERIAL NUMBER 14470876
	TUF key id: eaf22372f417dd618a46f6c627dbc276e9fd30a004fc94f9be946e73f8bd090b

VERIFIED KEY WITH SERIAL NUMBER 15938765
	TUF key id: f40f32044071a9365505da3d1e3be6561f6f22d0e60cf51df783999f6c3429cb

VERIFIED KEY WITH SERIAL NUMBER 15938791
	TUF key id: f505595165a177a41750a8e864ed1719b1edfccd5a426fd2c0ffda33ce7ff209

VERIFIED KEY WITH SERIAL NUMBER 18158855
	TUF key id: 75e867ab10e121fdef32094af634707f43ddd79c6bab8ad6c5ab9f03f4ea8c90

# To manually verify the chain for any key ID

	export SERIAL_NUMBER=${SERIAL_NUMBER}
	openssl verify -verbose -x509_strict -CAfile <(cat piv-attestation-ca.pem ceremony/2022-07-12/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_device_cert.pem) ceremony/2022-07-12/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_key_cert.pem

# Manually extract the public key for any key ID and match with published

	export SERIAL_NUMBER=${SERIAL_NUMBER}
	openssl x509 -in ceremony/2022-07-12/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_key_cert.pem -pubkey -noout
	cat ceremony/2022-07-12/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_pubkey.pem
+ '[' -f /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12/staged/root.json ']'
+ ./verify repository --repository /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12 --staged
STAGED METADATA

Outputting metadata verification at /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12...

Verifying staging.json...
	Success! Signatures valid and threshold achieved
	staging version 2, expires 2022/11/10

Verifying snapshot.json...
	Success! Signatures valid and threshold achieved
	snapshot version 38, expires 2022/08/03

Verifying revocation.json...
	Success! Signatures valid and threshold achieved
	revocation version 1, expires 2023/01/12

Verifying timestamp.json...
	Success! Signatures valid and threshold achieved
	timestamp version 38, expires 2022/07/27

Verifying targets.json...
	Success! Signatures valid and threshold achieved
	targets version 4, expires 2023/01/12

Verifying rekor.json...
	Success! Signatures valid and threshold achieved
	rekor version 3, expires 2023/01/12

Verifying root.json...
	Success! Signatures valid and threshold achieved
	root version 4, expires 2023/01/12
+ '[' -f /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12/repository/1.root.json ']'
+ ./verify repository --repository /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12 --root /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12/repository/1.root.json

VERIFYING TUF CLIENT UPDATE

Client successfully initialized, updating and downloading targets...
Client updated to...
	root.json version 3, expires 2022/11/10
	timestamp.json version 37, expires 2022/07/22
	snapshot.json version 37, expires 2022/07/29
	targets.json version 3, expires 2022/11/10

Retrieved target ctfe.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEbfwR+RJudXscgRBRpKX1XFDy3Pyu
dDxz/SfnRi1fT8ekpfBd2O1uoz7jr3Z8nKzxA69EUQ+eFCFI3zeubPWU7w==
-----END PUBLIC KEY-----

Retrieved target fulcio.crt.pem...
-----BEGIN CERTIFICATE-----
MIIB+DCCAX6gAwIBAgITNVkDZoCiofPDsy7dfm6geLbuhzAKBggqhkjOPQQDAzAq
MRUwEwYDVQQKEwxzaWdzdG9yZS5kZXYxETAPBgNVBAMTCHNpZ3N0b3JlMB4XDTIx
MDMwNzAzMjAyOVoXDTMxMDIyMzAzMjAyOVowKjEVMBMGA1UEChMMc2lnc3RvcmUu
ZGV2MREwDwYDVQQDEwhzaWdzdG9yZTB2MBAGByqGSM49AgEGBSuBBAAiA2IABLSy
A7Ii5k+pNO8ZEWY0ylemWDowOkNa3kL+GZE5Z5GWehL9/A9bRNA3RbrsZ5i0Jcas
taRL7Sp5fp/jD5dxqc/UdTVnlvS16an+2Yfswe/QuLolRUCrcOE2+2iA5+tzd6Nm
MGQwDgYDVR0PAQH/BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQEwHQYDVR0OBBYE
FMjFHQBBmiQpMlEk6w2uSu1KBtPsMB8GA1UdIwQYMBaAFMjFHQBBmiQpMlEk6w2u
Su1KBtPsMAoGCCqGSM49BAMDA2gAMGUCMH8liWJfMui6vXXBhjDgY4MwslmN/TJx
Ve/83WrFomwmNf056y1X48F9c4m3a3ozXAIxAKjRay5/aj/jsKKGIkmQatjI8uup
Hr/+CxFvaJWmpYqNkLDGRU+9orzh5hI2RrcuaQ==
-----END CERTIFICATE-----

Retrieved target fulcio_v1.crt.pem...
-----BEGIN CERTIFICATE-----
MIIB9zCCAXygAwIBAgIUALZNAPFdxHPwjeDloDwyYChAO/4wCgYIKoZIzj0EAwMw
KjEVMBMGA1UEChMMc2lnc3RvcmUuZGV2MREwDwYDVQQDEwhzaWdzdG9yZTAeFw0y
MTEwMDcxMzU2NTlaFw0zMTEwMDUxMzU2NThaMCoxFTATBgNVBAoTDHNpZ3N0b3Jl
LmRldjERMA8GA1UEAxMIc2lnc3RvcmUwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAT7
XeFT4rb3PQGwS4IajtLk3/OlnpgangaBclYpsYBr5i+4ynB07ceb3LP0OIOZdxex
X69c5iVuyJRQ+Hz05yi+UF3uBWAlHpiS5sh0+H2GHE7SXrk1EC5m1Tr19L9gg92j
YzBhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRY
wB5fkUWlZql6zJChkyLQKsXF+jAfBgNVHSMEGDAWgBRYwB5fkUWlZql6zJChkyLQ
KsXF+jAKBggqhkjOPQQDAwNpADBmAjEAj1nHeXZp+13NWBNa+EDsDP8G1WWg1tCM
WP/WHPqpaVo0jhsweNFZgSs0eE7wYI4qAjEA2WB9ot98sIkoF3vZYdd3/VtWB5b9
TNMea7Ix/stJ5TfcLLeABLE4BNJOsQ4vnBHJ
-----END CERTIFICATE-----

Retrieved target rekor.0.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
-----END PUBLIC KEY-----

Retrieved target rekor.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
-----END PUBLIC KEY-----

Retrieved target artifact.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhyQCx0E9wQWSFI9ULGwy3BuRklnt
IqozONbbdbqz11hlRJy9c7SG+hdcFl9jE9uE/dwtuwU2MqU9T/cN0YkWww==
-----END PUBLIC KEY-----
```

```
kommendorkapten@m1m14:~/git/root-signing % GITHUB_USER=kommendorkapten ./verify repository --repository=$(pwd)/ceremony/2022-07-12 --staged
STAGED METADATA

Outputting metadata verification at /Users/kommendorkapten/git/root-signing/ceremony/2022-07-12...

Verifying rekor.json...
	Success! Signatures valid and threshold achieved
	rekor version 3, expires 2023/01/12

Verifying staging.json...
	Success! Signatures valid and threshold achieved
	staging version 2, expires 2022/11/10

Verifying root.json...
	Success! Signatures valid and threshold achieved
	root version 4, expires 2023/01/12

Verifying targets.json...
	Success! Signatures valid and threshold achieved
	targets version 4, expires 2023/01/12

Verifying revocation.json...
	Success! Signatures valid and threshold achieved
	revocation version 1, expires 2023/01/12

Verifying timestamp.json...
	Success! Signatures valid and threshold achieved
	timestamp version 38, expires 2022/07/27

Verifying snapshot.json...
	Success! Signatures valid and threshold achieved
	snapshot version 38, expires 2022/08/03
```

**Comment by mihaimaruseac** [comment]
```
Verifying snapshot.json...
	Success! Signatures valid and threshold achieved
	snapshot version 38, expires 2022/08/03

Verifying timestamp.json...
	Success! Signatures valid and threshold achieved
	timestamp version 38, expires 2022/07/27
```

```
VERIFYING TUF CLIENT UPDATE

Client successfully initialized, updating and downloading targets...
Client updated to...
	timestamp.json version 37, expires 2022/07/22
	snapshot.json version 37, expires 2022/07/29
	targets.json version 3, expires 2022/11/10
	root.json version 3, expires 2022/11/10
```

**Review by mihaimaruseac — APPROVED** [review]
Hashes valid too

```
mihaimaruseac@ankh:~/src/repos/gosst/root-signing$ sha256sum $(pwd)/ceremony/2022-07-12/staged/targets.json
730ce11e76f9c91f3814556f97f84566d95f6908b246a0013d7d66db52167117  /usr/local/google/home/mihaimaruseac/src/repos/gosst/root-signing/ceremony/2022-07-12/staged/targets.json
mihaimaruseac@ankh:~/src/repos/gosst/root-signing$ cat $(pwd)/ceremony/2022-07-12/staged/snapshot.json | jq -r '.signed.meta."targets.json"'
{
  "hashes": {
    "sha256": "730ce11e76f9c91f3814556f97f84566d95f6908b246a0013d7d66db52167117",
    "sha512": "7c6d8beacd11aa9011a700a0509fead75fcf54d836a414f5633a2c29ee3c10415797a38aa85396953a3c6b206a052225852ce2cc85ae8b110d849855d32d0f59"
  },
  "length": 5984,
  "version": 4
}
```

**Comment by Hayden-IO** [comment]
```
Outputting metadata verification at /usr/local/google/home/hblauzvern/root-signing/ceremony/2022-07-12...

Verifying targets.json...
	Success! Signatures valid and threshold achieved
	targets version 4, expires 2023/01/12

Verifying timestamp.json...
	Success! Signatures valid and threshold achieved
	timestamp version 38, expires 2022/07/27

Verifying snapshot.json...
	Success! Signatures valid and threshold achieved
	snapshot version 38, expires 2022/08/03

Verifying rekor.json...
	Success! Signatures valid and threshold achieved
	rekor version 3, expires 2023/01/12

Verifying revocation.json...
	Success! Signatures valid and threshold achieved
	revocation version 1, expires 2023/01/12

Verifying root.json...
	Success! Signatures valid and threshold achieved
	root version 4, expires 2023/01/12

Verifying staging.json...
	Success! Signatures valid and threshold achieved
	staging version 2, expires 2022/11/10
```

**Review by Hayden-IO — APPROVED** [review]

**Review by Hayden-IO — CHANGES_REQUESTED** [review]
Offline comment about rekor.0.pub in targets, want to check if we're ok with that

**Comment by asraa** [comment]
> Offline comment about rekor.0.pub in targets, want to check if we're ok with that

i think it's fine. it's the same key as rekor.pub

**Review by Hayden-IO — APPROVED** [review]

**Comment by asraa** [comment]
Merging, I'll be leaving the final publishing PR open for longer, and you can still do hash verification!

**Pull request merged**